lib/ui/compositing.dart

```diff
@@ -219,68 +230,8 @@ class ShaderMaskEngineLayer extends _EngineLayerWrapper {
 /// To draw graphical operations onto a [Scene], first create a
 /// [Picture] using a [PictureRecorder] and a [Canvas], and then add
 /// it to the scene using [addPicture].
-class SceneBuilder extends NativeFieldWrapperClass1 {
-  /// Creates an empty [SceneBuilder] object.
-  @pragma('vm:entry-point')
-  SceneBuilder() {
-    _constructor();
-  }
-
-  @Native<Void Function(Handle)>(symbol: 'SceneBuilder::Create')
-  external void _constructor();
-
-  // Layers used in this scene.
-  //
-  // The key is the layer used. The value is the description of what the layer
-  // is used for, e.g. "pushOpacity" or "addRetained".
-  final Map<EngineLayer, String> _usedLayers = <EngineLayer, String>{};
-
-  // In debug mode checks that the `layer` is only used once in a given scene.
-  bool _debugCheckUsedOnce(EngineLayer layer, String usage) {
-    assert(() {
-      assert(
-          !_usedLayers.containsKey(layer),
-          'Layer ${layer.runtimeType} already used.\n'
-          'The layer is already being used as ${_usedLayers[layer]} in this scene.\n'
-          'A layer may only be used once in a given scene.');
-
-      _usedLayers[layer] = usage;
-      return true;
-    }());
-
-    return true;
-  }
-
-  bool _debugCheckCanBeUsedAsOldLayer(_EngineLayerWrapper? layer, String methodName) {
-    assert(() {
-      if (layer == null) {
-        return true;
-      }
-      assert(layer._nativeLayer != null, 'Object disposed');
-      layer._debugCheckNotUsedAsOldLayer();
-      assert(_debugCheckUsedOnce(layer, 'oldLayer in $methodName'));
-      layer._debugWasUsedAsOldLayer = true;
-      return true;
-    }());
-    return true;
-  }
-
-  final List<_EngineLayerWrapper> _layerStack = <_EngineLayerWrapper>[];
-
-  // Pushes the `newLayer` onto the `_layerStack` and adds it to the
-  // `_debugChildren` of the current layer in the stack, if any.
-  bool _debugPushLayer(_EngineLayerWrapper newLayer) {
-    assert(() {
-      if (_layerStack.isNotEmpty) {
-        final _EngineLayerWrapper currentLayer = _layerStack.last;
-        currentLayer._debugChildren ??= <_EngineLayerWrapper>[];
-        currentLayer._debugChildren!.add(newLayer);
-      }
-      _layerStack.add(newLayer);
-      return true;
-    }());
-    return true;
-  }
+abstract class SceneBuilder {
+  factory SceneBuilder() => _NativeSceneBuilder();
```

Comment:
Here and elsewhere - should this be a redirecting factory instead, i.e.

```suggestion
  factory SceneBuilder() = _NativeSceneBuilder;
```

Reply:
done